XRPLF/rippled
# Added ResourceReport message.

## High Level Overview of Change

Implemented according to:
https://gist.github.com/RichardAH/0ee2c651093d2ca2fa8a0484292990ec

This change allows peers to send, receive, and log resource usage statistics of other peers.

### Context of Change

Diagnostics and troubleshooting improvement.

### Type of Change


## Before / After
After this change, the ripple configuration can be adjusted to enable the resource_report:

```
[resource_report]
1
```

Once enabled, debug log files may contain:

`2022-Jan-16 13:44:14.740510620 UTC Protocol:DBG [001] Received ResourceReport for [016] : Idle_perc: 78.2841797; FDSize_rippled: 64; LoadAvg_5min: 1.11000001; SwapFree_kb: 33518332; SwapTotal_kb: 33520636; LoadAvg_15min: 0.850000024; Uptime_h: 483.333344; MemAvailable_kb: 64439300; Uptime_rippled_h: 0.0166666675; VmSwap_rippled_kb: 0; LoadAvg_1min: 3.21000004; Cpu_rippled_perc: 91.6722488; VmSize_rippled_kb: 2206480; num_threads_rippled: 24; MemFree_kb: 38513104; MemTotal_kb: 65638480;
`

## Test Plan

- Run two instances of rippled and make sure these nodes are connected to each other by specifying the [ips_fixed] in the rippled configuration file. Inspect the logfile to ensure that the resource report is logged (see above).
- Execute peermon (https://github.com/RichardAH/xrpl-peermon) and verify that messages with type mtResourceReport are received.

## Future Tasks

Potentially add additional useful resource metrics?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203452398484741
